pkg/cmd/pulumi/destroy.go

```diff
@@ -210,5 +211,11 @@ func newDestroyCmd() *cobra.Command {
 			&eventLogPath, "event-log", "",
 			"Log events to a file at this path")
 	}
+
+	// internal flag
+	cmd.PersistentFlags().StringVar(&execKind, "exec-kind", "", "")
+	// ignore err, only happens if flag does not exist
+	_ = cmd.PersistentFlags().MarkHidden("exec-kind")
```

Comment:
Any benefit to making this a flag vs just an env var?

Seems an env var might offer more flexibility?

Also - any reason to constrain the values vs just allow consuming code to set this to whatever they want?

Reply:
This is not an attempt to design an extensible feature. The flag (implementation details) is hidden and can always be swapped out in the future if we get feedback from console users that they'd like to customize this value.